File: BestTimeToBuyAndSellStockII_122.java
class Solution {
    public static int maxProfit(int[] prices) {
        int buy = prices[0];
        int size = prices.length;
        int sumProfit = 0;
        for(int i = 0; i  < size - 1; i++) {
            if(prices[i] > prices[i+1]) {
                sumProfit += prices[i] - buy;
                buy = prices[i+1];
            }
        }
        sumProfit += prices[size - 1] - buy;
        return sumProfit;
    }
    
    public static void main(String args[]) {
        int[] nums = {7,1,5,3,6,4};
        System.out.println(maxProfit(nums));
    }
}

File: BestTimeToBuyAndSellStockII_122.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        buyPrice = prices[0]
        size = len(prices)
        profit = 0
        for i in range (0,size - 1) :
            if prices[i] > prices[i + 1] :
                profit += prices[i] - buyPrice
                buyPrice = prices[i + 1]
        profit += prices[size - 1] - buyPrice
        return profit

if __name__ == '__main__':
    prices = [7,1,5,3,6,4]
    solu = Solution()
    print(solu.maxProfit(prices))